Specs/blob/blob-storage-2017-07-29.json

```diff
@@ -2059,10 +2059,10 @@
             "$ref": "#/parameters/IfNoneMatch"
           },
           {
-            "$ref": "#/parameters/BlobContentLength"
+            "$ref": "#/parameters/BlobContentLengthOptional"
```

Comment:
Should we be making many of the other parameters optional for the other APIs?

Reply:
Are there other parameters we should be making required then? For example some of the lease operations would require lease ids.

Reply:
Yes, you're right. A couple of the lease API parameters we should make optional/remove. Let's create another task for that